File: test_cbunu.py
import requests
s = requests.Session()
s.headers.update({'User-Agent': 'Mozilla/5.0'})
# First GET to establish PHPSESSID
r1 = s.get('https://cbunu.com/truyen-tranh/end-scoop-199-chap-78.html')
print('GET1 cookies:', s.cookies.get_dict())
# POST password
r2 = s.post('https://cbunu.com/truyen-tranh/end-scoop-199-chap-78.html', data={'access_pass': '12345'}, allow_redirects=False)
print('POST cookies:', s.cookies.get_dict())
# GET page again
r3 = s.get('https://cbunu.com/truyen-tranh/end-scoop-199-chap-78.html')
print('GET2 cookies:', s.cookies.get_dict())
print('GET2:', r3.status_code, '<title>Đăng nhập</title>' in r3.text)

File: test_cbunu.lua
package.path = '/mnt/d/Project/truyenfull/truyenviet.koplugin/?.lua;' .. package.path
local Http = require('truyenviet.http_client')

local url = 'https://cbunu.com/truyen-tranh/end-scoop-199-chap-78.html'
local body = 'access_pass=12345'
local post_headers = {
    ['Content-Type'] = 'application/x-www-form-urlencoded',
    ['Referer'] = 'https://cbunu.com/',
    ['Origin'] = 'https://cbunu.com',
}

local html, err, headers, code, resp_body = Http:request('POST', url, body, post_headers, { redirect = false })
print('CODE:', code)
if headers then
    for k, v in pairs(headers) do print(k, v) end
end
